### PR TITLE
Feature/display and save video

### PR DESCRIPTION
**Pull request by ismailIBNMAJAH**
## Overview
This PR introduces the video storage and display features for EduVerse, enhancing the media experience within the app by enabling users to save, post, and view videos with a vertical scrolling interface.

## Key Features
- Video storage functionality to securely save user-generated content
- Save and post functionalities for both photos and videos
- Vertical scrolling in VideoScreen for seamless navigation through media
- MVVM architecture for organized publication management
- Firebase Storage integration for media files (photos and videos)
- Full test coverage for UI interactions and functionalities

##Technical Implementation
- Architecture: MVVM pattern ensuring clean separation of concerns
- State Management: Kotlin Flow for reactive updates and seamless UI transitions
- Database: Firebase Firestore for managing publication data
- Storage: Firebase Storage for both photos and videos
- UI: Jetpack Compose with Material3 for consistent design
Testing: Comprehensive UI and functional tests with Android test frameworks and mocked Firebase dependencies

## Testing
- UI tests covering all major interactions, including video scrolling, saving, and posting
- Tests for handling loading, success, and error states
- Mocked Firebase instances to simulate data interactions
- Achieved test coverage > ?

## Screenshots
Here is a preview of the VideoScreen with vertical scrolling:
tiktokWhileScrolling

## Related Issues
- Implements video storage and publication features
- Addresses user needs for media display and sharing
- Testing Instructions
- Run the app and navigate to the VideoScreen.
- Test the save functionality for both photos and videos.
- Scroll through videos to confirm smooth vertical navigation.
- Check loading and error handling states during posting and viewing. 

## Notes
- Requires Firebase setup for Storage and Firestore
- Ensure appropriate indexing in Firestore for optimal performance

### CODE REVIEW

File: app/src/main/java/com/github/se/eduverse/repository/PublicationRepository.kt
Comment (Fayroth) [inline app/src/main/java/com/github/se/eduverse/repository/PublicationRepository.kt:1]: LGTM

File: app/src/main/java/com/github/se/eduverse/repository/VideoRepository.kt
Comment (Fayroth) [inline app/src/main/java/com/github/se/eduverse/repository/VideoRepository.kt:1]: Good implementation of the functions overall, nice job. (dont forget to comment in english)

File: app/src/main/java/com/github/se/eduverse/ui/camera/PicTakenScreen.kt
Comment (Fayroth) [inline app/src/main/java/com/github/se/eduverse/ui/camera/PicTakenScreen.kt:1]: LGTM

File: app/src/main/java/com/github/se/eduverse/ui/camera/nextScreen.kt
Comment (Fayroth) [inline app/src/main/java/com/github/se/eduverse/ui/camera/nextScreen.kt:1]: Nice implementation of the publish logic, looks good, seperated the case of photo and video, handles failure, good job !

File: app/src/main/java/com/github/se/eduverse/ui/videos/Videos.kt
Comment (Fayroth) [inline app/src/main/java/com/github/se/eduverse/ui/videos/Videos.kt:1]: Good implementation of the scroll feature, smooth and working.

File: app/src/main/java/com/github/se/eduverse/viewmodel/PublicationViewModel.kt
Comment (Fayroth) [inline app/src/main/java/com/github/se/eduverse/viewmodel/PublicationViewModel.kt:1]: I feel like we dont need the second function, we could just use the first one with some conditions and maybe changing its parameter but its ok, the actual implementations works well.

File: app/src/main/java/com/github/se/eduverse/viewmodel/VideoViewModel.kt
Comment (Fayroth) [inline app/src/main/java/com/github/se/eduverse/viewmodel/VideoViewModel.kt:1]: LGTM

**Review by Fayroth — APPROVED** [review]
# Pull Request Review: Video Storage & Display Features

## General Overview
The implementation of video features and UI shows good understanding of the architecture and proper use of Firebase integration. Code quality is generally high with good separation of concerns.

## Detailed Review

### `MainActivity.kt`
❌ Issue with EduverseApp composable parameter handling
* Dependencies should be initialized inside the composable
* Current approach works but isn't best practice
* Consider refactoring for future implementations

### `IVideoRepository.kt`
⚠️ Minor concern
* updateVideo function seems unnecessary
* Not blocking, but worth reconsidering in future iterations

### `VideoRepository.kt`
✅ Strong implementation
* Functions well implemented
* Good overall structure
* Note: Please use English for comments

### `nextScreen.kt` (Camera)
✅ Excellent implementation
* Clean publish logic separation between photo/video
* Good failure handling
* Well-structured code

### `Videos.kt`
✅ High quality implementation
* Smooth scroll functionality
* Well-executed video display features

### `PublicationViewModel.kt`
⚠️ Suggestion
* Second function might be redundant
* Could be consolidated with first function using conditional logic
* Current implementation works well regardless

## Recommendations
1. Refactor EduverseApp composable initialization
2. Add English comments in VideoRepository
3. Consider consolidating PublicationViewModel functions in future updates

## Conclusion
✅ PR is ready to merge after addressing the composable initialization concern
* Code quality is high
* Core functionality is well implemented
* Minor improvements can be addressed in future iterations

File: app/src/main/java/com/github/se/eduverse/MainActivity.kt
[the remark this answers is not in the document]
Comment (ismailIBNMAJAH) [reply app/src/main/java/com/github/se/eduverse/MainActivity.kt:1]: you're right, i'm sorry about that. i'll fix that.

**Pull request merged**